packages/dds/map/src/test/mocha/fuzzUtils.ts

```diff
@@ -0,0 +1,511 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import {
+	type AsyncGenerator,
+	type AsyncReducer,
+	type Generator,
+	combineReducers,
+	combineReducersAsync,
+	createWeightedAsyncGenerator,
+	createWeightedGenerator,
+	takeAsync,
+} from "@fluid-private/stochastic-test-utils";
+import type { Client, DDSFuzzModel, DDSFuzzTestState } from "@fluid-private/test-dds-utils";
+import type { IFluidHandle } from "@fluidframework/core-interfaces";
+import { isObject } from "@fluidframework/core-utils/internal";
+import type { Serializable } from "@fluidframework/datastore-definitions/internal";
+import { isFluidHandle } from "@fluidframework/runtime-utils/internal";
+
+import {
+	DirectoryFactory,
+	type IDirectory,
+	type ISharedMap,
+	MapFactory,
+} from "../../index.js";
+
+import { assertEquivalentDirectories } from "./directoryEquivalenceUtils.js";
+
+/**
+ * Represents a map clear operation.
+ */
+interface MapClear {
+	type: "clear";
+}
+
+/**
+ * Represents a map set key operation.
+ */
+interface MapSetKey {
+	type: "setKey";
+	key: string;
+	value: Serializable<unknown>;
+}
+
+/**
+ * Represents a map delete key operation.
+ */
+interface MapDeleteKey {
+	type: "deleteKey";
+	key: string;
+}
+
+type MapOperation = MapSetKey | MapDeleteKey | MapClear;
+
+// This type gets used a lot as the state object of the suite; shorthand it here.
+type MapState = DDSFuzzTestState<MapFactory>;
+
+async function assertMapsAreEquivalent(a: ISharedMap, b: ISharedMap): Promise<void> {
+	assert.equal(a.size, b.size, `${a.id} and ${b.id} have different number of keys.`);
+	for (const key of a.keys()) {
+		const aVal: unknown = a.get(key);
+		const bVal: unknown = b.get(key);
+		if (isObject(aVal) === true) {
+			assert(
+				isObject(bVal),
+				`${a.id} and ${b.id} differ at ${key}: a is an object, b is not}`,
+			);
+			const aHandle = isFluidHandle(aVal) ? await aVal.get() : aVal;
+			const bHandle = isFluidHandle(bVal) ? await bVal.get() : bVal;
```

Comment:
I noticed you converted comparison to shallow equality based on absolute paths in directory. You probably need the same thing here, right?